### ISSUE TITLE
Can we have hover actions on acronyms and key words?

### ISSUE THREAD

**Issue by gregmacfarlane** [tfresource/website#261]
The style guide will ask that people properly define acronyms. We were wondering if it would be possible to have some sort of hover tool linked to a glossary. When people hover over a highlighted word, their tooltip would then show the definition of that acronym or term.

**Comment by billyc** [comment]
I think yes it's possible? Just guessing here -- haven't looked into it
deeply.

We would need to (1) write a small markdown plugin which would get parsed
during page build. So something like

*:acronym[TRB|Transportation Research Board]:*

and then (2) write a Vue component to pop up a custom element on hover.

..b

**Comment by billyc** [comment]
@gregmacfarlane - I looked into this a bit more. Short answer is "yes".

- We could maintain one file with all definitions, let's say `glossary.yml`
- One page on the site would list all of them in a nicely formatted index/glossary
- When you want to reference a definition with a pop-up hover in a content page, the author would include some sort of tag (to be determined) -- something like `::acronym{TRB}::`
- The page build process would turn that tag into the proper HTML/CSS for the popup automatically

There's some coding to be done, but not much, to make this happen. Let me know.

**Comment by gregmacfarlane** [comment]
Yes, let's get that started

**Comment by billyc** [comment]
I have the tooltip working now.  See
https://tfresource.org/topics/

for an example.

To add an acronym, you need to do two things:
  - Add its definition to `/glossary.txt`
  - Reference it on any page using `==TRB==` etc.

@gregmacfarlane : I haven't created the glossary page yet. Before I do, a question: are simple ACRONYM: Description fields enough (e.g. "TRB: Transportation Research Board"), or do you think for some definitions people might want to put more explanation text? If yes, it would just require the format of the `glossary.txt` file to something more structured.

**Comment by gregmacfarlane** [comment]
I think it looks really good, although it doesn't appear functional in my locally served version yet. I think it would be good to limit explanatory text to a link to a page or a subsection where the acronym can be more adequately described.

**Comment by billyc** [comment]
yes, whoops -- the way I built it, the acronyms are replaced during the website build process, not during the "hot-reload" of the local developer build. I will fix that but might take a while before I get to that. Hopefully this is good enough for now.

I added a first cut of the glossary page. Acronyms can now include a link for more info.
https://tfresource.org/topics/glossary

**NOTE** The acronym definitions have moved to `.vuepress/public/acronyms.csv`
